source/frontend/src/features/properties/map/compensation/CompensationRequisitionTrayContainer.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCompensationRequisitionRepository } from 'hooks/repositories/useRequisitionCompensationRepository';
+import { Api_AcquisitionFile } from 'models/api/AcquisitionFile';
+import { Api_Compensation } from 'models/api/Compensation';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { SystemConstants, useSystemConstants } from 'store/slices/systemConstants';
+
+import { SideBarContext } from '../context/sidebarContext';
+import { CompensationRequisitionTrayViewProps } from './CompensationRequisitionTrayView';
+
+export interface ICompensationRequisitionTrayContainerProps {
+  compensationRequisitionId?: number;
+  onClose: () => void;
+  View: React.FunctionComponent<React.PropsWithChildren<CompensationRequisitionTrayViewProps>>;
+}
+
+export const CompensationRequisitionTrayContainer: React.FunctionComponent<
+  React.PropsWithChildren<ICompensationRequisitionTrayContainerProps>
+> = ({ compensationRequisitionId, onClose, View }) => {
+  const { getSystemConstant } = useSystemConstants();
+  const { file } = useContext(SideBarContext);
+
+  const [editMode, setEditMode] = useState(false);
+  const [loadedCompensation, setLoadedCompensation] = useState<Api_Compensation | undefined>();
+
+  const clientConstant = getSystemConstant(SystemConstants.CLIENT);
+  const gstConstant = getSystemConstant(SystemConstants.GST);
+  const gstDecimal = gstConstant !== undefined ? parseFloat(gstConstant.value) : undefined;
+
+  const {
+    getCompensationRequisition: { execute: getCompensationRequisition, error, loading },
+  } = useCompensationRequisitionRepository();
+
+  const fetchCompensationReq = useCallback(async () => {
+    if (!!compensationRequisitionId) {
```

Comment:
what about if the compensationRequisitionId changes from the currently loaded compensationReq?

Reply:
the whole tray gets reloaded when router detect changes. so far that's the only way to update the id.